### ISSUE TITLE
Returning data in dictionary format

### ISSUE THREAD

**Issue by tiferrei** [nerdishbynature/RequestKit#57]
Recently I've been trying to come up with a way to let users access not only the pretty object parsed data, but also more raw data like the the returned JSON in dictionary format. I was hoping to be able to do something like this:
```swift
public func rooms(_ session: RequestKitURLSession = URLSession.shared, roomID: String = "", roomName: String = "", siteID: String = "", siteName: String = "", classification: Classification = Classification.Unknown, capacity: String = "", _ inDictionary: Bool = false, completion: @escaping (_ response: Response<Any>) -> Void) -> URLSessionDataTaskProtocol? {
        let router = RoomBookingsRouter.readRooms(configuration, roomID, roomName, siteID, siteName, classification, capacity)
        if inDictionary {
            return router.load(session, expectedResultType: [String: Any].self) {
            // ...
        }
        return router.load(session, expectedResultType: RoomsResponse.self) { rooms, error in
           // ...
        }
    }
```

However, due to having the completion defined in the function header, I need to specify a return type.

Any ideas on how to better approach this?

Thanks a lot,
Tiago

**Comment by tiferrei** [comment]
@pietbrauer Maybe an even better way would be allowing RequestKit based framework access to the `Data` itself?

**Comment by tiferrei** [comment]
I've found a better (ish?) way of converting this, basically a `toDictionary()` method that extends `Encodable`:
```swift
extension Encodable {
    public func toDictionary() -> [String: Any] {
        let encoder = JSONEncoder()
        do {
            let data = try encoder.encode(self)
            if let dict = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] {
                return dict
            }
        } catch {
            return ["error": "UCLKit failed to encode object."]
        }
        return ["error": "UCLKit failed to encode object."]
    }
}
```

It works great, and the performance is way better than my previous solution. However, I notice that the `CodingKeys` in the dictionary are the ones from the original JSON, instead of the object properties.

**Comment by tiferrei** [comment]
For those interested this is what I ended up with:
```swift
extension UCLResponse {
    /**
     - Parses an UCLResponse conforming object into a `[String: Any]` dictionary.
     - ⚠️: This method will return the data using **UCL API's original keys**.
     - The reason for a `[String: Any]` is that this method is meant mainly to display raw UCL API data.
     - returns: A `[String: Any]` dictionary representing the object's data.
     */
    public func toDictionary() -> [String: Any] {
        let encoder = JSONEncoder()
        if #available(iOS 10.0, OSX 10.12, tvOS 10.0, watchOS 3.0, *) {
            encoder.dateEncodingStrategy = .iso8601
        } else {
            encoder.dateEncodingStrategy = .deferredToDate
        }
        do {
            let data = try encoder.encode(self)
            if let dict = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] {
                return dict
            }
        } catch {
            return ["error": "UCLKit failed to encode object."]
        }
        return ["error": "UCLKit failed to encode object."]
    }
}
```
`UCLResponse` being a `Codable` conforming protocol to whom root level response classes must conform.
